keras/src/layers/core/embedding.py

```diff
@@ -388,25 +372,26 @@ def quantize(self, mode):
                 lambda shape, dtype: embeddings_scale,
             )
         else:
-            raise NotImplementedError(
-                self.QUANTIZATION_MODE_ERROR_TEMPLATE.format(mode=mode)
-            )
+            raise self._quantization_mode_error(mode)
         self._tracker.lock()
 
         # Set new dtype policy
-        if not isinstance(
-            self.dtype_policy, dtype_policies.QuantizedDTypePolicy
-        ):
-            quantized_dtype = f"{mode}_from_{self.dtype_policy.name}"
+        if not self.dtype_policy.is_quantized:
             # We set the internal `self._dtype_policy` instead of using the
             # setter to avoid double `quantize` call
-            self._dtype_policy = dtype_policies.get(quantized_dtype)
+            policy = dtype_policies.get(f"{mode}_from_{self.dtype_policy.name}")
```

Comment:
Can we actually make this go through the layer `dtype_policy` setter?

```python
policy = dtype_policies.get(f"{mode}_from_{self.dtype_policy.name}")
self.dtype_policy = policy
```

I would be really great if layer implementations (dense/embedding/etc) had no idea about `DTypePolicyMap` class, and the map stays a superclass implementation detail. That would allow people to write much simple layer subclasses.

Seems like we could do that, we add an early return at the top of this method?

```python
if self._is_quantized:
    return
```

Maybe there's a better way to do this, but any way we could get the next 6 lines to an implementation details of the base layer that would probably be a good thing.

Reply:
I agree with you. It is feasible, and I have made the update.